.circleci/config.yml

```diff
@@ -25,30 +25,33 @@ runOnDocker: &runOnDocker
   working_directory: /go/src/github.com/mesg-foundation/core
 
 jobs:
-  "build_docker":
-    <<: *runOnMachine
+  "unit_test":
+    <<: *runOnDocker
     steps:
       - checkout
-      - run: docker pull mesg/core:latest
-      - run: docker build -t mesg/core:$CIRCLE_SHA1 --build-arg version="dev build `echo $CIRCLE_SHA1 | cut -c1-7`" .
-
-  "test":
+      - setup_remote_docker # this should be remove when no test are actually using docker
+      - run: go get -t ./...
+      - run: docker swarm init # this should be remove when no test are actually using docker
+      - run: docker build -t sleep docker-images/sleep/ # this should be remove when no test are actually using docker
+      - run: docker build -t http-server docker-images/http-server/ # this should be remove when no test are actually using docker
+      - run: env MESG_CORE_IMAGE=mesg/core:NONEXISTINGIMAGE go test -timeout 180s -p 1 -coverprofile=coverage.txt ./...
+      - run: bash <(curl -s https://codecov.io/bash)
```

Comment:
We're not using `codecov` - maybe think of removing it and include when we actually start using this in our dev pipeline

Reply:
I would like to keep it as a reminder of the code coverage stats.
It's true we don't enforce it strictly, but still we should have the stats in the PR.